### ISSUE TITLE
Wrong R.pick typings

### ISSUE THREAD

**Issue by selfrefactor** [selfrefactor/rambda#462]
https://github.com/selfrefactor/rambda/issues/459#issuecomment-634870095

**Comment by selfrefactor** [comment]
Thank you @noamcore for bringing the issue up. I moved to separate issue, so we can discuss in separate thread.

Your initial example won't work with `Ramda` as well, as you need to pass the custom type to `R.pick`.

This will work:

```typescript
describe('R.pick bug', () => {
  type MyObject = {
    id?: number;
    type: string;
    value: string;
  }
  const myObj: MyObject = { id: 0, type: 'classA', value: 'foo' };

  it('happy', () => {
    const result = pick<MyObject, string>(['type', 'value'], myObj);
    result // $ExpectType Pick<MyObject, "type" | "value" | "id">
    result.id // $ExpectType number | undefined
    result.type // $ExpectType string
  })
})
```

I will publish later today a patch version of `Rambda` with correct typings.

**Comment by selfrefactor** [comment]
Fixed with `5.4.2` but feel free to comment even if the issue is closed.
